File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/Link.java
package org.metabrainz.mobile.data.sources.api.entities;

public class Link {
    private String type;
    private Url url;
    private boolean ended;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Url getUrl() {
        return url;
    }

    public void setUrl(Url url) {
        this.url = url;
    }

    public boolean isEnded() {
        return ended;
    }

    public void setEnded(boolean ended) {
        this.ended = ended;
    }

    public String getPageTitle() {
        String resource = url.getResource().trim();
        return resource.substring(resource.lastIndexOf("/") + 1);
    }
}

File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/Link.kt
package org.metabrainz.mobile.data.sources.api.entities

class Link {
    var type: String? = null
    var url: Url? = null
    var isEnded = false
    val pageTitle: String
        get() {
            val resource = url!!.resource!!.trim { it <= ' ' }
            return resource.substring(resource.lastIndexOf("/") + 1)
        }
}